app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,51 @@
+class PostsController < ApplicationController
+
+  def index
+    @posts = Post.all
+    @comments = Comment.all
+  end
+
+  def show
+    @posts = Post.all
+    @post = Post.find(params[:id])
+    @comments = Comment.all
+    @comment = Comment.new
+  end
+
+  def new
+    @post = Post.new
+    # @comment = Comment.new
+  end
+
+  def edit
+    @post = Post.find(params[:id])
+  end
+
+  def update
+    @post = Post.find(params[:id])
+    @post.update(post_params)
+    redirect_to @post
+  end
+
+  def create
+    @post = Post.create(post_params)
+    @comment = Comment.create(@comment)
```

Comment:
as a matter of separation of concerns and single responsibility. The create action should only create 1 model, the model the controller is based around. In this case, the posts controller should only create posts.